Document tagged reductions (#5)
kokkos/kokkos-core-wiki: API-Reference/Kokkos::parallel_reduce.md

  * If `ExecPolicy` is `MDRangePolicy` the `functor` has a member function of the form `operator() (const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const` or `operator() (const WorkTag, const IntegerType& i0, ... , const IntegerType& iN, ReducerValueType& value) const` 
    * The `WorkTag` free form of the operator is used if `ExecPolicy::work_tag` is not `void`.
    * `N` must match `ExecPolicy::rank`

>   * If the `functor` is a lambda, `ReducerArgument` must satisfy the `Reducer` concept or `ReducerArgumentNonConst` must be a POD type with `operator +=` and `operator =` or a `Kokkos::View`.  In the latter case, the default `Sum` reduction is applied. In this case, the potential `init`/`join`/`final` member functions must be untagged even for tagged reductions.

Reviewer: "untagged" does not seem to be defined.
```suggestion
  * If the `functor` is a lambda, `ReducerArgument` must satisfy the `Reducer` concept or `ReducerArgumentNonConst` must be a POD type with `operator +=` and `operator =` or a `Kokkos::View`.  In the latter case, the default `Sum` reduction is applied. If provided, the `init`/`join`/`final` member functions do not take a `WorkTag` argument.
```
